runtime/src/impls.rs

```diff
@@ -81,14 +81,22 @@ impl Convert<u128, Balance> for CurrencyToVoteHandler {
 
 /// Convert from weight to balance via a simple coefficient multiplication
 /// The associated type C encapsulates a constant in units of balance per weight
-pub struct LinearWeightToFee<C>(sp_std::marker::PhantomData<C>);
+pub struct ScaleLinearWeightToFee<C>(sp_std::marker::PhantomData<C>);
 
-impl<C: Get<Balance>> Convert<Weight, Balance> for LinearWeightToFee<C> {
+impl<C: Get<Balance>> Convert<Weight, Balance> for ScaleLinearWeightToFee<C> {
 	fn convert(w: Weight) -> Balance {
 		// cennznet-node a weight of 10_000 (smallest non-zero weight) to be mapped to 10^7 units of
 		// fees, hence:
 		let coefficient = C::get();
-		Balance::from(w).saturating_mul(coefficient)
+		let linear_weight = Balance::from(w).saturating_mul(coefficient);
+		// Scale down fees 10^-12
+		let fee_divider: Balance = 1_000_000_000_000;
+		let scaled_weight = linear_weight.checked_div(fee_divider).unwrap_or(linear_weight);
+		if scaled_weight.is_zero() {
+			linear_weight
+		} else {
+			scaled_weight
```

Comment:
This is probably not a good idea. Let me give you an example. Suppose the scale down factor is 7. If the fee is 8 the result will be 1, If the fee is 6 the result will be 6. It means the fees now have lost their proportions.

Reply:
I get your point.. what if I have min_weight..
if scaled_weight.is_zero() && linear_weight < min_weight {
  linear_weight
} else if scaled_weight.is_zero() {
 min_weight
} else {
 scaled_weight
}